### ISSUE TITLE
[Feature Request]Add setting to make certain branches read-only (disable commiting)

### ISSUE THREAD

**Issue by VolkmarR** [ForkIssues/TrackerWin#1003]
Since we use the Release Flow in Azure Devops, we have disabled the ability to push to master. Now it sometimes happens, that I accidentally make a commit on master. 

It would be very useful, if I could tell Fork to disable the commit ability for some branches (in my case master) and therefor protect them from accidentally commiting to them.

**Comment by DanPristupov** [comment]
Currently you can't do this.

As a quick workaround you can create the following pre-push hook in git (I didn't test it). It will not allow you to push to `$protected_branch`

`.git/hooks/pre-push`:

```
#!/bin/sh

protected_branch='master'
current_branch=$(git symbolic-ref HEAD | sed -e 's,.*/\(.*\),\1,')

if [ $protected_branch = $current_branch ]
then
    exit 1 # push will not execute
else
    exit 0 # push will execute
fi
```

**Comment by VolkmarR** [comment]
Thanks for the fast response. If I understand this correctly, I have to implement a pre-commit hook instead of a pre-push hook, because I wan't to block commits on master. I'll check this out.

**Comment by DanPristupov** [comment]
Yes. Something like following (https://stackoverflow.com/a/40465455):

`.git/pre-commit`

```
#!/bin/sh

protected_branch='master'
branch="$(git rev-parse --abbrev-ref HEAD)"

if [ $branch = $protected_branch ]; then
  echo "You can't commit directly to master branch"
  exit 1
fi
```
